src/main/java/org/springframework/shell2/JLineShell.java

```diff
@@ -155,10 +165,7 @@ public void run() throws IOException {
 					result = ReflectionUtils.invokeMethod(method, methodTarget.getBean(), args);
 				}
 				catch (ExitRequest e) {
-					if (applicationContext instanceof Closeable) {
```

Comment:
I don't know if we're saying the same thing, but yes, I want to make this pluggable (if only to allow unit testing this new behavior).
Could you elaborate on your differentiation between `ExitRequest` and `ExitResult`? (Do you mean use the `ResultHandler` mechanism?